bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/ProperMotion.scala

```diff
@@ -11,16 +15,126 @@ import Scalaz._
 case class ProperMotion(
   deltaRA: RightAscensionAngularVelocity,
   deltaDec: DeclinationAngularVelocity,
-  epoch: Epoch = Epoch.J2000)
+  epoch: Epoch = Epoch.J2000) {
+
+  /** Coordinates at the current time. */
+  def calculateAt(t: Target, i: Instant): Coordinates =
+    plusYears(t, epoch.untilInstant(i))
+
+  /** Coordinates `elapsedYears` fractional epoch-years after `epoch`. */
+  def plusYears(t: Target, elapsedYears: Double): Coordinates = {
+    val baseCoordinates = t.coords(None).getOrElse(Coordinates.zero)
+
+    // TODO: Is this calculated correctly?
```

Comment:
I'm not convinced that I'm handling the `properVelocity` correctly. @swalker2m, could you check?

Reply:
This looks right, but again we should add sanity checks.